### PR TITLE
Add support for XEP-0319: Last User Interaction

### PR DESCRIPTION
**Pull request by vanitasvitae**
This PR adds Elements and Providers for [XEP-0319: Last User Interaction in Presence](https://xmpp.org/extensions/xep-0319.html).

### CODE REVIEW

**Comment by vanitasvitae** [comment]
@Flowdalic this PR only adds the elements and providers for the XEP. What do you think of also adding a manager class which has a timer. That timer can be reset by the user by calling `.activity()` or something. Once that method did not get called for more than 5 mins or so (configurable), the timer fires and a presence stanza with an idle element will be sent automatically.

**Comment by vanitasvitae** [comment]
Another idea would be to create methods to automatically get the last user interaction of contacts by caching last interactions in a map.

**Comment by vanitasvitae** [comment]
Resolved conflicts.

**Comment by Flowdalic** [comment]
The commit message is missing the issue key (SMACK-636). Please add it.

File: smack-extensions/src/main/java/org/jivesoftware/smackx/last_interaction/provider/IdleProvider.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.jivesoftware.smack.provider.ExtensionElementProvider;
+import org.jivesoftware.smackx.last_interaction.element.IdleElement;
+
+import org.jxmpp.util.XmppDateTime;
+import org.xmlpull.v1.XmlPullParser;
+
+public class IdleProvider extends ExtensionElementProvider<IdleElement> {
+
+    public static IdleProvider TEST_INSTANCE = new IdleProvider();
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/last_interaction/provider/IdleProvider.java]: Nit: Missing `final` keyword

**Comment by vanitasvitae** [comment]
Done.

**Pull request merged**